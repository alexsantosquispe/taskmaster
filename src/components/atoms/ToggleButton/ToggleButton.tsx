import type { ReactNode } from 'react';

interface ToggleButtonProps {
  isCollapsed: boolean;
  setIsCollapsed: (value: boolean) => void;
  expandIcon: ReactNode;
  collapseIcon: ReactNode;
}

export const ToggleButton = ({
  isCollapsed,
  setIsCollapsed,
  expandIcon,
  collapseIcon
}: ToggleButtonProps) => {
  const toggleButton = () => {
    setIsCollapsed(!isCollapsed);
  };

  return (
    <button
      aria-label={isCollapsed ? 'Expand menu' : 'Collapse menu'}
      className="rounded-lg p-2 text-neutral-600 hover:cursor-pointer hover:bg-neutral-100 hover:text-blue-700 dark:text-white/60 dark:hover:bg-neutral-800 dark:hover:text-orange-500"
      onClick={toggleButton}
    >
      {isCollapsed ? expandIcon : collapseIcon}
    </button>
  );
};
