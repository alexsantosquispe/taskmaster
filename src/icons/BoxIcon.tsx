import { twMerge } from 'tailwind-merge';
import type { SvgIconProps } from '../models/types';

export const BoxIcon = ({ className }: SvgIconProps) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 16 16"
      fill="none"
      id="Box--Streamline-Majesticons"
      height={16}
      width={16}
      className={twMerge('size-6', className)}
    >
      <desc>{'\n    Box Streamline Icon: https://streamlinehq.com\n  '}</desc>
      <path
        fill="currentColor"
        fillRule="evenodd"
        d="M7.6466666666666665 1.4346666666666668a0.6666666666666666 0.6666666666666666 0 0 1 0.7066666666666667 0l4.602666666666666 2.876666666666667L8 7.226666666666667 3.0439999999999996 4.311333333333333l4.602666666666666 -2.876666666666667zM2.005333333333333 5.247333333333334A0.6673333333333332 0.6673333333333332 0 0 0 2 5.333333333333333v5.333333333333333a0.6666666666666666 0.6666666666666666 0 0 0 0.3133333333333333 0.5653333333333332L7.333333333333333 14.369333333333332v-5.9879999999999995L2.005333333333333 5.246666666666666zM8.666666666666666 14.369333333333332l5.02 -3.1373333333333333A0.6666666666666666 0.6666666666666666 0 0 0 14 10.666666666666666V5.333333333333333c0 -0.028666666666666663 -0.002 -0.057999999999999996 -0.005333333333333333 -0.086L8.666666666666666 8.380666666666666v5.988666666666667z"
        clipRule="evenodd"
        strokeWidth={0.6667}
      />
    </svg>
  );
};
