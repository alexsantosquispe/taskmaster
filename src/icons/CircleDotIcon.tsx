import type { SvgIconProps } from '../models/types';
import { twMerge } from 'tailwind-merge';

export const CircleDotIcon = ({ className }: SvgIconProps) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
      className={twMerge('size-5', className)}
    >
      <circle cx="12" cy="12" r="10" />
      <circle cx="12" cy="12" r="1" />
    </svg>
  );
};
