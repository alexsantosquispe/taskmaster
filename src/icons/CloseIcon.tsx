import { twMerge } from 'tailwind-merge';
import type { SvgIconProps } from '../models/types';

export const CloseIcon = ({ className }: SvgIconProps) => {
  return (
    <svg
      data-testid="close-icon"
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
      className={twMerge('size-5', className)}
    >
      <path d="M18 6 6 18" />
      <path d="m6 6 12 12" />
    </svg>
  );
};
