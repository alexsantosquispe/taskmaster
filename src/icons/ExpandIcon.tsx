import { twMerge } from 'tailwind-merge';
import type { SvgIconProps } from '../models/types';

export const ExpandIcon = ({ className }: SvgIconProps) => {
  return (
    <svg
      data-testid="expand-icon"
      xmlns="http://www.w3.org/2000/svg"
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
      className={twMerge('size-5', className)}
    >
      <rect width="18" height="18" x="3" y="3" rx="2" />
      <path d="M15 3v18" />
      <path d="m8 9 3 3-3 3" />
    </svg>
  );
};
