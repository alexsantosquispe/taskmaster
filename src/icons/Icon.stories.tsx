import * as Icons from './index';

import type { Meta } from '@storybook/react';
import { useState } from 'react';

export default {
  title: 'Icons'
} as Meta;

export const IconGallery = () => {
  const [filter, setFilter] = useState('');

  const filteredIcons = Object.entries(Icons).filter(([name]) =>
    name.toLowerCase().includes(filter.toLowerCase())
  );

  return (
    <div className="space-y-6 p-4">
      <div className="flex w-1/2 flex-col gap-y-1">
        <label>Icons Gallery</label>
        <input
          placeholder="Search icons..."
          value={filter}
          onChange={(e) => setFilter(e.target.value)}
          className="rounded-md border border-neutral-300 p-2"
        />
      </div>

      <div className="flex flex-wrap gap-4">
        {filteredIcons.map(([name, IconComponent]) => (
          <div
            key={name}
            className="flex h-[10rem] w-[10rem] flex-col items-center justify-center gap-4 rounded-lg border border-neutral-300"
          >
            <IconComponent className="size-6" />
            <pre className="text-primary text-xs font-medium">{name}</pre>
          </div>
        ))}
      </div>
    </div>
  );
};
