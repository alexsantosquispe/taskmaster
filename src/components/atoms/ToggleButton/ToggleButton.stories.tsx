import type { Meta, StoryObj } from '@storybook/react';

import { useState } from 'react';
import { CollapseIcon, ExpandIcon } from '../../../icons';
import { ToggleButton } from './ToggleButton';

const meta = {
  title: 'Atoms/ToggleButton',
  component: ToggleButton,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof ToggleButton>;

export default meta;

type Story = StoryObj<typeof ToggleButton>;

export const Default: Story = () => {
  const [isCollapsed, setIsCollapsed] = useState(false);

  return (
    <ToggleButton
      isCollapsed={isCollapsed}
      setIsCollapsed={setIsCollapsed}
      expandIcon={<ExpandIcon />}
      collapseIcon={<CollapseIcon />}
    />
  );
};

Default.args = {};
