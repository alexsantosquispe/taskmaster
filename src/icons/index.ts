import { BoardIcon } from './BoardIcon';
import { BoxIcon } from './BoxIcon';
import { CalendarIcon } from './CalendarIcon';
import { ChartPieIcon } from './ChartPieIcon';
import { ChevronDownIcon } from './ChevronDownIcon';
import { ChevronUpIcon } from './ChevronUpIcon';
import { CircleCheckIcon } from './CircleCheckIcon';
import { CircleDashedIcon } from './CircleDashedIcon';
import { CircleDotDashedIcon } from './CircleDotDashedIcon';
import { CirclePlusIcon } from './CirclePlusIcon';
import { CollapseIcon } from './CollapseIcon';
import { EllipsisVerticalIcon } from './EllipsisVerticalIcon';
import { ExpandIcon } from './ExpandIcon';
import { InfoIcon } from './InfoIcon';
import { LaptopIcon } from './LaptopIcon';
import { LoadIcon } from './LoadIcon';
import { MoonIcon } from './MoonIcon';
import { ProjectIcon } from './ProjectIcon';
import { SettingsIcon } from './SettingsIcon';
import { SunIcon } from './SunIcon';
import { TasksListIcon } from './TasksListIcon';

export {
  BoardIcon,
  BoxIcon,
  CalendarIcon,
  ChartPieIcon,
  ChevronDownIcon,
  ChevronUpIcon,
  CircleCheckIcon,
  CircleDashedIcon,
  CircleDotDashedIcon,
  CirclePlusIcon,
  CollapseIcon,
  EllipsisVerticalIcon,
  ExpandIcon,
  InfoIcon,
  LaptopIcon,
  LoadIcon,
  MoonIcon,
  ProjectIcon,
  SettingsIcon,
  SunIcon,
  TasksListIcon
};
