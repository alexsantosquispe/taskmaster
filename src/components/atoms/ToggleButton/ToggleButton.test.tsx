import { fireEvent, render, screen } from '@testing-library/react';
import { CollapseIcon, ExpandIcon } from '../../../icons';

import { ToggleButton } from './ToggleButton';

describe('ToggleButton', () => {
  const setIsCollapsedMock = jest.fn();
  const props = {
    isCollapsed: false,
    setIsCollapsed: setIsCollapsedMock,
    expandIcon: <ExpandIcon />,
    collapseIcon: <CollapseIcon />
  };

  it('should render the component correctly', () => {
    let component = render(<ToggleButton {...props} />);
    expect(component).toMatchSnapshot();

    component = render(<ToggleButton {...props} isCollapsed={true} />);
    expect(component).toMatchSnapshot();
  });

  it('should toggle between icons', () => {
    render(<ToggleButton {...props} />);

    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(setIsCollapsedMock).toHaveBeenCalledTimes(1);
    expect(setIsCollapsedMock).toHaveBeenCalledWith(true);
  });
});
