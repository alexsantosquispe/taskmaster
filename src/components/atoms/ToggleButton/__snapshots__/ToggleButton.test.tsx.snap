// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ToggleButton should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="Collapse menu"
        class="rounded-lg p-2 text-neutral-600 hover:cursor-pointer hover:bg-neutral-100 hover:text-blue-700 dark:text-white/60 dark:hover:bg-neutral-800 dark:hover:text-orange-500"
      >
        <svg
          class="size-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="3"
          />
          <path
            d="M15 3v18"
          />
          <path
            d="m8 9 3 3-3 3"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="Collapse menu"
      class="rounded-lg p-2 text-neutral-600 hover:cursor-pointer hover:bg-neutral-100 hover:text-blue-700 dark:text-white/60 dark:hover:bg-neutral-800 dark:hover:text-orange-500"
    >
      <svg
        class="size-5"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="18"
          rx="2"
          width="18"
          x="3"
          y="3"
        />
        <path
          d="M15 3v18"
        />
        <path
          d="m8 9 3 3-3 3"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ToggleButton should render the component correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-label="Collapse menu"
        class="rounded-lg p-2 text-neutral-600 hover:cursor-pointer hover:bg-neutral-100 hover:text-blue-700 dark:text-white/60 dark:hover:bg-neutral-800 dark:hover:text-orange-500"
      >
        <svg
          class="size-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="3"
          />
          <path
            d="M15 3v18"
          />
          <path
            d="m8 9 3 3-3 3"
          />
        </svg>
      </button>
    </div>
    <div>
      <button
        aria-label="Expand menu"
        class="rounded-lg p-2 text-neutral-600 hover:cursor-pointer hover:bg-neutral-100 hover:text-blue-700 dark:text-white/60 dark:hover:bg-neutral-800 dark:hover:text-orange-500"
      >
        <svg
          class="size-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="3"
          />
          <path
            d="M15 3v18"
          />
          <path
            d="m10 15-3-3 3-3"
          />
        </svg>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-label="Expand menu"
      class="rounded-lg p-2 text-neutral-600 hover:cursor-pointer hover:bg-neutral-100 hover:text-blue-700 dark:text-white/60 dark:hover:bg-neutral-800 dark:hover:text-orange-500"
    >
      <svg
        class="size-5"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="18"
          rx="2"
          width="18"
          x="3"
          y="3"
        />
        <path
          d="M15 3v18"
        />
        <path
          d="m10 15-3-3 3-3"
        />
      </svg>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
